File: 12.375- Adivina El numero/CS/AdivinaElNumeroCS/AdivinaElNumeroCS/Form1.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace AdivinaElNumeroCS
{
	public partial class frmAdivinaElNumeroCS : Form
	{
		public frmAdivinaElNumeroCS()
		{
			InitializeComponent();
		}

		private void frmAdivinaElNumeroCS_Load(object sender, EventArgs e)
		{

		}


		private string number;
		private bool proceso = false;

		public void Preguntar()
		{
			proceso = true;
			VBMath.Randomize();
			number = Conversions.ToString(Convert.ToInt32(VBMath.Rnd() * 10));
			while (true)
			{
				if ((Interaction.InputBox("Que numero estoy pensando?") ?? "") == (number ?? ""))
				{
					MessageBox.Show("Correcto, adivina otro numero");
					proceso = false;
					break;
				}
				else
				{
					MessageBox.Show("No, intenta de nuevo");
				}
			}
		}

		private void tmrPrincipal_Tick(object sender, EventArgs e)
		{
			if (proceso == false)
			{
				Preguntar();
			}
		}
	}
}

File: 12.375- Adivina El numero/vb/KaboomVB/KaboomVB/Form1.vb
Public Class frmKaboomVB

    Dim number As String
    Dim proceso As Boolean = False



    Sub Preguntar()
        proceso = True
        Randomize()
        number = Convert.ToInt32(Rnd() * 10)

        While (True)

            If (InputBox("Que numero estoy pensando?") = number) Then
                MsgBox("Correcto, adivina otro numero")
                proceso = False
                Exit While
            Else
                MsgBox("No, intenta de nuevo")
            End If

        End While

    End Sub









    'Genera numero aleatorio
    Private Sub frmKaboomVB_Load(sender As Object, e As EventArgs) Handles MyBase.Load


    End Sub





    Private Sub tmrPrincipal_Tick(sender As Object, e As EventArgs) Handles tmrPrincipal.Tick
        If proceso = False Then

            Preguntar()
        End If

    End Sub

End Class
